x/src/cargo.rs

```diff
@@ -86,6 +99,17 @@ impl Cargo {
         self
     }
 
+    pub fn env_additions<I, S>(&mut self, args: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<str>,
```

Comment:
To make your life easier (and to completely avoid the "to_map" function) I recommend you change this function to look closer to the one in the stdlib, no need to reinvent the wheel
```suggestion
pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
where
    I: IntoIterator<Item = (K, V)>,
    K: AsRef<OsStr>,
    V: AsRef<OsStr>, 
```

Reply:
This was the function I was fighting with last night.    I tried 

I: IntoIterator<Item = (String, String)>,

And  IntoIterator<Item = (&str, &str)>,

But when I was storing HashMap<&str, &str> in the enumeration the enumeration wasn't able to derive the Default trait.   Tried String and I think it didn't have the Clone trait, might have that flipped?   Hard to remember -- was pretty frantic.   I'll give this a shot soon, but will wait to do that kind of refactoring until after feature complete and unblock fuzzing coverage.

Reply:
Yeah i would probably not store these args in the CargoArgs struct (which needs to be Default) but in the CargoCommand enum which doesn't need to impl any of that

Reply:
thanks, that helped a bit.   I was more fighting traits than the borrow logic.   but the combo wasn't a good time.